src/test/java/uk/gov/pay/api/json/CreatePaymentRequestDeserializerTest.java

```diff
@@ -624,19 +504,18 @@ public void shouldDeserializeARequestWithPrefilledCardholderDetailsSuccessfully(
                 "\"country\": \"GB\"\n" +
                 "}" + "}" + "}";
 
-        ValidCreatePaymentRequest paymentRequest = deserializer.deserialize(jsonFactory.createParser(payload), ctx);
+        CreatePaymentRequest paymentRequest = deserializer.deserialize(jsonFactory.createParser(payload), ctx);
         assertThat(paymentRequest.getAmount(), is(1000));
         assertThat(paymentRequest.getReference(), is("Some reference"));
         assertThat(paymentRequest.getDescription(), is("Some description"));
-        assertThat(paymentRequest.getReturnUrl().get(), is("https://somewhere.gov.uk/rainbow/1"));
-        assertThat(paymentRequest.getLanguage(), is(Optional.empty()));
-        assertThat(paymentRequest.getDelayedCapture(), is(Optional.empty()));
-        assertThat(paymentRequest.getEmail().get(), is("j.bogs@example.org"));
-        assertThat(paymentRequest.getPrefilledCardholderDetails().isPresent(), is(true));
-        assertThat(paymentRequest.getPrefilledCardholderDetails().get().getCardholderName().isPresent(), is(true));
-        assertThat(paymentRequest.getPrefilledCardholderDetails().get().getCardholderName().get(), is("J Bogs"));
-        assertThat(paymentRequest.getPrefilledCardholderDetails().get().getBillingAddress().isPresent(), is(true));
-        Address billingAddress = paymentRequest.getPrefilledCardholderDetails().get().getBillingAddress().get();
+        assertThat(paymentRequest.getReturnUrl(), is("https://somewhere.gov.uk/rainbow/1"));
+        assertThat(paymentRequest.getLanguage(), is(nullValue()));
+        assertThat(paymentRequest.getDelayedCapture(), is(nullValue()));
+        assertThat(paymentRequest.getEmail(), is("j.bogs@example.org"));
+        assertThat(paymentRequest.getPrefilledCardholderDetails().getCardholderName().isPresent(), is(true));
```

Comment:
It’s… _interesting_ that we’ve ended up with an object graph where top-level getters like `getLanguage()` can return null but deeper getters like `getCarholderName()` return `Optional`. It would be nice to make this consistent.

Reply:
Yes, I'll fix this up as part of this story. Probably towards the end once the restructuring of validation is complete to avoid too much change on change.